punish/punish.py

```diff
@@ -1304,6 +1387,7 @@ def execute_unpunish(self, server_id, member_id) -> bool:
             member_data = data.get(member.id, {})
             caseno = member_data.get('caseno')
             mod = self.bot.get_cog('Mod')
+            removed_roles = member_data.get('removed_roles')
```

Comment:
```suggestion
            removed_roles = member_data.get('removed_roles', [])
```
This needs a default to handle an absent key for existing cases.